util/git/git.go

```diff
@@ -45,6 +45,16 @@ func SameURL(leftRepo, rightRepo string) bool {
 // Prefer using SameURL() over this function when possible. This algorithm may change over time
 // and should not be considered stable from release to release
 func NormalizeGitURL(repo string) string {
+	return normalizeGitURL(repo, false)
+}
+
+// Similar to NormalizeGitURL, except returning a not fully normalized string when the url is invalid.
+// Needed to allow a deletion of repos with invalid urls. See https://github.com/argoproj/argo-cd/issues/20921.
+func NormalizeGitURLAllowInvalid(repo string) string {
+	return normalizeGitURL(repo, true)
```

Comment:
I think it will be better not to add allowInvalid parameter to private function. Can we just return original "repo" in case if error happens ? In such case you will not need to modify with additional conditions normalizeGitURL function

Reply:
I've made the change.